### DISCUSSION TITLE
trying to reset locked accounts on mealie nighly with script

### DISCUSSION THREAD

**Discussion by freebs65** [mealie-recipes/mealie#4320] [General]
Using Mealie nightly .. I"m trying to use  /app/mealie/scripts/reset_locked_users.py

It starts then fails with raise ValueError("group_id must be set")..

so i tried change password script as well.. same result..

any thoughts on this ? Do I need to add something to command line as an option ?

Thanks!

**Comment by cw-cdz** [comment]
I am seeing this exact same issue.
I looked at the code for reset_locked_users, and it takes no parameters.
I traced the error through, but not being familiar with the code I was unable to determine exactly what possible group ids there are. Hopefully someone who knows the architecture a little more can chime in. 
I'll try to dig further this weekend when I have some time.

```python
Traceback (most recent call last):
  File "/app/mealie/scripts/reset_locked_users.py", line 34, in <module>
    main()
  File "/app/mealie/scripts/reset_locked_users.py", line 20, in main
    locked_users = user_service.get_locked_users()
  File "/app/mealie/services/user_services/user_service.py", line 14, in get_locked_users
    return self.repos.users.get_locked_users()
  File "/usr/local/lib/python3.10/functools.py", line 981, in __get__
    val = self.func(instance)
  File "/app/mealie/repos/repository_factory.py", line 173, in users
    return RepositoryUsers(self.session, PK_ID, User, PrivateUser, group_id=self.group_id)
  File "/app/mealie/repos/repository_generic.py", line 470, in __init__
    raise ValueError("group_id must be set")
ValueError: group_id must be set
```

**Reply by hay-kot** [reply]
This was recently fixed, try pulling the latest nightly and giving it another go.

**Comment by mckochan** [comment]
still getting this error on latest release (not nightly)

**Reply by michael-genson** [reply]
Are you sure you're on 2.1? Can you post your settings page and logs?

**Reply by mckochan** [reply]
I can't get to the settings page as I can't login. I just pulled latest release, so it must be 2.1
Only error in log is:
```
ERROR    2024-11-01T10:05:21 - User is locked out from 192.168.1.15
INFO     2024-11-01T10:05:21 - [192.168.1.15:58476] 423 Locked "POST /api/auth/token HTTP/1.1
```

**Reply by michael-genson** [reply]
This is saying that you're locked out. Have you run the unlock script?

**Reply by mckochan** [reply]
Yes and I am getting the error that this thread is on. Here is full log when running the script:
```
# python /app/mealie/scripts/reset_locked_users.py
Are you sure you want to reset all locked users? (y/n) y
Traceback (most recent call last):
  File "/app/mealie/scripts/reset_locked_users.py", line 34, in <module>
    main()
  File "/app/mealie/scripts/reset_locked_users.py", line 20, in main
    locked_users = user_service.get_locked_users()
  File "/app/mealie/services/user_services/user_service.py", line 14, in get_locked_users
    return self.repos.users.get_locked_users()
  File "/usr/local/lib/python3.10/functools.py", line 981, in __get__
    val = self.func(instance)
  File "/app/mealie/repos/repository_factory.py", line 173, in users
    return RepositoryUsers(self.session, PK_ID, User, PrivateUser, group_id=self.group_id)
  File "/app/mealie/repos/repository_generic.py", line 470, in __init__
    raise ValueError("group_id must be set")
ValueError: group_id must be set
```

**Reply by michael-genson** [reply]
Ah I see, this is a different issue than what was fixed. I'll get a PR in. Thanks!

**Reply by adam57jerome** [reply]
moi c'est pareil, j'ai le même message que @mckochan  "j'espère que dans 24 heures, je pourrais y accéder

**Reply by adam57jerome** [reply]
Same here, I have the same message as @mckochan: 'I hope that in 24 hours, I will be able to access it.

**Reply by michael-genson** [reply]
Will be fixed by: https://github.com/mealie-recipes/mealie/pull/4487
Alternatively you can edit the database manually and remove the lock.

**Reply by adam57jerome** [reply]
Oui, mais là, je ne sais pas faire.
Yes, but right now, I don't know how to do it.
